### PR TITLE
Project Desc & User Groups for PDF Report

### LINKED ISSUE #20: High-level project description with correct identification of user groups
**Describe what needs to get done**
Explain how each user group is expected to use the software differently. (Recall that in order for two users to be considered as being in two different user groups, they must use the system differently either via a functional or non-functional system requirement.)

**Describe the solution you'd like**

This section **must** be included in finalized PDF report.

### PR DESCRIPTION
**Pull request by vnajack**
closes #20

### CODE REVIEW

File: docs/reports/requirements/1_proj-desc.md [hunk 1]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
 
 ## PDF Report
 
-I will write my PDF report section here to later be incorporated into the full document.
+### High-level Description (about one paragraph)
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:65]:
```suggestion
### High-level Description
```
Is the "about one paragraph" portion meant to remain in the final PDF report?

File: docs/reports/requirements/1_proj-desc.md [hunk 2]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+
+The objective for the Speaking Portal Project is to design, develop, and deploy a lip-sync animation (LSA) API for the
+Kukarella Table Reads text-to-speech (TTS) web application. This API will serve as an animation generating add-on for
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:68]: Maybe I'm misunderstanding, but I believe it's not necessarily meant to be a part of the table reads system, but rather its own kukarella plugin

File: docs/reports/requirements/1_proj-desc.md [hunk 3]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+this system so that the user can both listen to and watch their avatar speak the user-provided text.
+
+### User groups
+
+In this project, there are two main user groups who will be using the software: general users and developers. The
+general users are the general public who use the Kukarella web application; this could be anyone who would like to
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:74]: It feels like a little too much of the use general in that first lil section, could maybe use a little rephrasing

File: docs/reports/requirements/1_proj-desc.md [hunk 4]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+listen to rather than read some text in any language with a customized avatar. The developers are the Kukarella
+employees who will be responsible for incorporating the API into their current system and adding more customizable
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:76]: This is not necessary at all just a lil simpler put:
```suggestion
listen to rather than read some text in any language with a customized avatar. The developers are the Kukarella
employees responsible system maintenance and the addition of more customizable
```

File: docs/reports/requirements/1_proj-desc.md [hunk 5]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+assets for the avatars in the future.
+
+In the current Kukarella Table Reads TTS web application, the first step is for the user to create one or more actors.
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:79]: see my first comment, I know they mentioned the table read plugin, but I thought they were using that more as an example of what plugins already exist

File: docs/reports/requirements/1_proj-desc.md [hunk 6]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+The user gives the actor a name, selects a voice from a list of voices they can preview, selects a language from the
+list of approximately 150 languages, and selects or uploads an image of the actor. After creating the actor and
+entering the text they want to listen to, the user presses the play button, and then waits for the application to
+process and generate the audio, which is then auto-played once the audio is complete.
+
+Once our API is incorporated into this system, if the user would like to both watch and listen to their avatar speak
+the provided text, the only difference in the user interface is that the user will select one of the "premium" avatar
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:86]: Do we know this will be the only difference at this step, since we aren't handling any UI we might not need to mention this

File: docs/reports/requirements/1_proj-desc.md [hunk 7]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+images for their actor. After pressing the play button, the user will wait for the Kukarella application and the API to
+process and generate a video of their avatar speaking the text with lips synchronized to the sounds produced in their
+generated speech.
+
+As for the developers, they will need to call the API within the Kukarella application. After receiving the user input
+and after the text is converted to speech in an audio file, the developer will call our API instead of providing the
+audio file as output for the user to listen to. When calling our API, the user-selected language, user-provided text,
+and the Kukarella-generated audio will be passed in, and then the API will provide the animated video as output. At
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:94]: Don't forget about the selected avatar!

File: docs/reports/requirements/1_proj-desc.md [hunk 8]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+this point, the Kukarella developer shows the video in the web application's user interface.
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:95]: I believe there is also an option to download

**Review by LoganParker — CHANGES_REQUESTED** [review]
Looks really good thank you Veronica! feel free to agree/disagree with my feedback as you see fit

File: docs/reports/requirements/1_proj-desc.md [hunk 9]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+
+The objective for the Speaking Portal Project is to design, develop, and deploy a lip-sync animation (LSA) API for the
+Kukarella Table Reads text-to-speech (TTS) web application. This API will serve as an animation generating add-on for
+this system so that the user can both listen to and watch their avatar speak the user-provided text.
+
+### User groups
+
+In this project, there are two main user groups who will be using the software: general users and developers. The
+general users are the general public who use the Kukarella web application; this could be anyone who would like to
+listen to rather than read some text in any language with a customized avatar. The developers are the Kukarella
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:75]:
```suggestion
have a more immersive text-to-speech experience where an animated avatar that reads aloud user text.
```

Sentence made need some re-word. I added a suggestion, but I think something else could be better too

File: docs/reports/requirements/1_proj-desc.md [hunk 10]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+employees who will be responsible for incorporating the API into their current system and adding more customizable
+assets for the avatars in the future.
+
+In the current Kukarella Table Reads TTS web application, the first step is for the user to create one or more actors.
+The user gives the actor a name, selects a voice from a list of voices they can preview, selects a language from the
+list of approximately 150 languages, and selects or uploads an image of the actor. After creating the actor and
+entering the text they want to listen to, the user presses the play button, and then waits for the application to
+process and generate the audio, which is then auto-played once the audio is complete.
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:83]:
```suggestion
process and generate the audio.
```

Might want to remove to avoid a run-on

File: docs/reports/requirements/1_proj-desc.md [hunk 11]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+
+Once our API is incorporated into this system, if the user would like to both watch and listen to their avatar speak
+the provided text, the only difference in the user interface is that the user will select one of the "premium" avatar
+images for their actor. After pressing the play button, the user will wait for the Kukarella application and the API to
+process and generate a video of their avatar speaking the text with lips synchronized to the sounds produced in their
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:88]:
```suggestion
process and generate a video of their avatar speaking the text with lips synchronized to the sounds produced from their text-to-speech API ```

File: docs/reports/requirements/1_proj-desc.md [hunk 12]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+generated speech.
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:89]:
```suggestion
```

File: docs/reports/requirements/1_proj-desc.md [hunk 13]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+
+As for the developers, they will need to call the API within the Kukarella application. After receiving the user input
+and after the text is converted to speech in an audio file, the developer will call our API instead of providing the
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:92]:
```suggestion
and after the text is converted to speech in an audio file, the developer will call our API.
```

Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:92]:
```suggestion
and after the text is converted to speech in an audio file, the developer will call our API.
```
I think the next bit would not be needed. Might sound smoother without.

File: docs/reports/requirements/1_proj-desc.md [hunk 14]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+audio file as output for the user to listen to. When calling our API, the user-selected language, user-provided text,
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:93]:
```suggestion
When calling our API, the user-selected language, user-provided text,
```
This change  will be required if changes are made in the above suggestion. Otherwise ignore.

**Review by edouarde1 — CHANGES_REQUESTED** [review]
Great work Veronica! You do a great job giving an overview of our animation add-on and how it integrates with the client't web app. I put in a couple suggestions, let me know if you need more clarification on them.

File: docs/reports/requirements/1_proj-desc.md (hunk 1 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:65]: I put it there to remind myself, but we don't want to keep it in the final report! Good catch!

File: docs/reports/requirements/1_proj-desc.md (hunk 2 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:68]: Ah, I think I remember Nazim showing us the Table Reads as an example, but I think you're right that it would be a possible thing just in general. I'll fix this

File: docs/reports/requirements/1_proj-desc.md (hunk 3 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:74]: Good catch, thanks! I'll adjust

File: docs/reports/requirements/1_proj-desc.md (hunk 6 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:86]: I think we need to say *something* because our app doesn't work for uploaded images

File: docs/reports/requirements/1_proj-desc.md (hunk 5 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:79]: I'll adjust, thanks!

File: docs/reports/requirements/1_proj-desc.md (hunk 8 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:95]: Good catch! I'll add that in

File: docs/reports/requirements/1_proj-desc.md (hunk 7 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:94]: Good catch! I'll add it in.

File: docs/reports/requirements/1_proj-desc.md (hunk 12 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:89]: I made adjustments to clarify this section, thanks!

File: docs/reports/requirements/1_proj-desc.md (hunk 14 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:93]: Removed, and will show next copy soon! Thanks for making it smoother!

File: docs/reports/requirements/1_proj-desc.md [hunk 15]
```diff
@@ -62,37 +62,36 @@ Note: the User section will be color-coded to show what's already in place and w
 
 ## PDF Report
 
-### High-level Description (about one paragraph)
+### High-level Description
 
 The objective for the Speaking Portal Project is to design, develop, and deploy a lip-sync animation (LSA) API for the
-Kukarella Table Reads text-to-speech (TTS) web application. This API will serve as an animation generating add-on for
+Kukarella text-to-speech (TTS) web application. This API will serve as an animation generating add-on for
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:68]: I like this a lot more

File: docs/reports/requirements/1_proj-desc.md [hunk 16]
```diff
@@ -62,37 +62,36 @@ Note: the User section will be color-coded to show what's already in place and w
...
-In this project, there are two main user groups who will be using the software: general users and developers. The
-general users are the general public who use the Kukarella web application; this could be anyone who would like to
-listen to rather than read some text in any language with a customized avatar. The developers are the Kukarella
-employees who will be responsible for incorporating the API into their current system and adding more customizable
-assets for the avatars in the future.
+In this project, there are two main user groups who will be using the software: users and developers. The users are the
+general public who use the Kukarella web application; this could be anyone who would like to have a more immersive
+text-to-speech experience where an animated avatar reads user text aloud. The developers are the Kukarella employees
+who will be responsible for incorporating the API into their current system and adding more customizable assets for the
+avatars in the future.
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:77]: this section flows nicely

File: docs/reports/requirements/1_proj-desc.md [hunk 17]
```diff
@@ -62,4 +62,38 @@ Note: the User section will be color-coded to show what's already in place and w
...
+### High-level Description
+
+The objective for the Speaking Portal Project is to design, develop, and deploy a lip-sync animation (LSA) API for the
+Kukarella Table Reads text-to-speech (TTS) web application. This API will serve as an animation generating add-on for
+this system so that the user can both listen to and watch their avatar speak the user-provided text.
+
+### User groups
+
+In this project, there are two main user groups who will be using the software: general users and developers. The
+general users are the general public who use the Kukarella web application; this could be anyone who would like to
```
Comment (mkuelker) [inline docs/reports/requirements/1_proj-desc.md]: Bit of a nitpick, but the _general_ users who are the _general_ public feels redundant

File: docs/reports/requirements/1_proj-desc.md [hunk 18]
```diff
@@ -62,37 +62,36 @@ Note: the User section will be color-coded to show what's already in place and w
...
-employees who will be responsible for incorporating the API into their current system and adding more customizable
-assets for the avatars in the future.
+In this project, there are two main user groups who will be using the software: users and developers. The users are the
+general public who use the Kukarella web application; this could be anyone who would like to have a more immersive
+text-to-speech experience where an animated avatar reads user text aloud. The developers are the Kukarella employees
+who will be responsible for incorporating the API into their current system and adding more customizable assets for the
+avatars in the future.
 
-In the current Kukarella Table Reads TTS web application, the first step is for the user to create one or more actors.
+In the current Kukarella TTS web application, the first step is for the user to create one or more actors.
```
Comment (LoganParker) [inline docs/reports/requirements/1_proj-desc.md:79]: This describes the table reads system still, but its a generally good description so it might not need to be changed

**Review by LoganParker — APPROVED** [review]
This looks great! Thank you for making those changes, I really like this :)

File: docs/reports/requirements/1_proj-desc.md [hunk 19]
```diff
@@ -65,34 +65,33 @@ Note: the User section will be color-coded to show what's already in place and w
...
-and after the text is converted to speech in an audio file, the developer will call our API instead of providing the
-audio file as output for the user to listen to. When calling our API, the user-selected language, user-provided text,
-and the Kukarella-generated audio will be passed in, and then the API will provide the animated video as output. At
-this point, the Kukarella developer shows the video in the web application's user interface.
+and after the text is converted to speech in an audio file, the developer will call our API. When calling our API, the
+user-selected actor image, the user-selected language, the user-provided text, and the Kukarella-generated audio will
+be passed in, and then the API will provide the animated video as output. At this point, the Kukarella developer shows
+the video in the web application's user interface and provides the option to download the video file.
 
 When Kukarella decides to add more "premium" avatars, the developer will need to include the head, mouth, eye, and
```
Comment (mkuelker) [inline docs/reports/requirements/1_proj-desc.md:96]: Premium needs to be changed to select

Comment (mkuelker) [inline docs/reports/requirements/1_proj-desc.md:96]: Change premium to select

**Review by mkuelker — CHANGES_REQUESTED** [review]
Just that little line fix

File: docs/reports/requirements/1_proj-desc.md [hunk 20]
```diff
@@ -62,4 +62,37 @@ Note: the User section will be color-coded to show what's already in place and w
...
+entering the text they want to listen to, the user presses the play button, and then waits for the application to
+process and generate the audio.
+
+Once our API is incorporated into this system, the enhanced experience of being able to both listen and watch the
+avatar speak will only be available with specific avatars. After pressing the play button, the user will wait for the
+Kukarella application and the API to process and generate a video of their avatar speaking the text with lips
+synchronized to the sounds produced in its speech.
+
+As for the developers, they will need to call the API within the Kukarella application. After receiving the user input
+and after the text is converted to speech in an audio file, the developer will call our API. When calling our API, the
```
Comment (edouarde1) [inline docs/reports/requirements/1_proj-desc.md:91]: The sentence that starts with "When.... " sounds a bit wordy.

Maybe could be split into two sentences ? 

".... wil be passed in as inputs. Then the API... "   <- This could be how you could split it. I am not sure if that flows better, but it could avoid a run-on

**Review by edouarde1 — APPROVED** [review]
I have put in one suggestion, but up to you if you would like to implement. Otherwise, it looks really good.

File: docs/reports/requirements/1_proj-desc.md (hunk 18 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:79]: I think I'll keep it because it does say, "one or more actors" which applies to both the general Kukarella app and the Table Reads, so I think it encompasses all "apps" within the product

File: docs/reports/requirements/1_proj-desc.md (hunk 20 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:91]: It does seem pretty long, so thank you for suggesting to split it up! Will do.

File: docs/reports/requirements/1_proj-desc.md (hunk 19 above)
Comment (vnajack) [reply docs/reports/requirements/1_proj-desc.md:96]: Good catch!! I forgot to adjust the last paragraph!

**Review by mkuelker — APPROVED** [review]
Looks Good!
